File: code_eval/java/moderate/Challenge46.java
package moderate;

import java.io.BufferedReader;
import java.io.File;
import java.io.FileReader;

/**
 * Challenge 46 : Primes less than X
 * Created by cschalk on 3/21/16.
 */
public class Challenge46 {

    public static void main (String[] args) throws Exception {
        File file = new File(args[0]);
        BufferedReader buffer = new BufferedReader(new FileReader(file));
        String line;
        while ((line = buffer.readLine()) != null) {
            listPrimesLessThan(line);
        }
    }

    private static void listPrimesLessThan(String input) {
        Long limit = Long.parseLong(input);
        StringBuilder primeList = new StringBuilder();
        if (limit > 2) {
            primeList.append("2");
        }
        for (long i = 3 ; i < limit ; i+=2) {
            if (isPrime(i)) {
                if (primeList.length() > 0) {
                    primeList.append(",");
                }
                primeList.append(i);
            }
        }

        System.out.println(primeList.toString());
    }

    private static boolean isPrime(long input) {
        if (input <= 1) { return false; }
        if (input == 2) { return true; }
        for (long i = 2 ; i < Math.sqrt(input) + 1 ; i++) {
            if (input % i == 0) {
                return false;
            }
        }
        return true;
    }
}

File: code_eval/cpp/moderate/challenge_46.cpp
// Created by cschalk on 3/30/16.
// Challenge 46 : Primes less than n
#include <iostream>
#include <fstream>
#include <cmath>

bool isPrime(int in) {
    switch (in) {
        case 1  : return false;
        case 2  : return true;
        default : {
            if (in % 2 == 0) {
                return false;
            }
            int limit = (int) std::sqrt(in) + 1;
            for (int i = 3 ; i < limit; i+=2) {
                if (in % i == 0) {
                    return false;
                }
            }
            return true;
        }
    }
}

int main(int argc, char ** argv) {
    std::ifstream input {argv[1]};
    int test;

    while (input >> test) {
        if (test >= 2) {
            std::cout << 2;
        }

        if (test > 2) {
            for (int i = 3 ; i < test ; i+=2) {
                if (isPrime(i)) {
                    std::cout << "," << i;
                }
            }
        }
        std::cout << std::endl;
    }
    return 0;
}
